### DISCUSSION TITLE
Custom REST endpoints

### DISCUSSION THREAD

**Discussion by BrettBedarf** [payloadcms/payload#532] [Q&A] [answered]
Is it possible to add custom REST endpoints?

**Comment by denolfe** [comment, marked as answer]
Hey @BrettBedarf, good question. Since Payload simply takes your express app as a parameter, you still have full access to it to add any additional functionality as needed. Here is an example `server.ts` that is generated running`npx create-payload-app`. You would simply add your additional functionality after the `payload.init` call.

```ts
import express from 'express';
import payload from 'payload';

require('dotenv').config();
const app = express();

// Initialize Payload
payload.init({
  secret: process.env.PAYLOAD_SECRET,
  mongoURL: process.env.MONGODB_URI,
  express: app,
})

// Add your own express routes here

app.listen(3000);
```

Hopefully, that answers your question. Let me know if you have any follow up questions 👍
